[Draft] Permanently align/anchor 2 objects (w/o parent-child relationship)

valeriosetti opened the pull request:
### Description of the feature or fix

When an object is aligned with `align_to()` then if the reference if moved, the aligned object will follow.

This PR also add an example making use of this feature: 1 parent object is created together with 3 followers. Then the parent is moved up & down on the screen and the followers follow it.

I marked as draft because this is my very first PR in LVGL so I would like to know whether this PR is moving in the right direction or not according to what it has been required in #5665.

Resolves #5665

liamHowatt commented on src/core/lv_obj_pos.c:
```diff
@@ -319,13 +319,26 @@ void lv_obj_align(lv_obj_t * obj, lv_align_t align, int32_t x_ofs, int32_t y_ofs
     lv_obj_set_pos(obj, x_ofs, y_ofs);
 }
 
+static void add_anchored_children(lv_obj_t * obj, const lv_obj_t * base)
```
Please rename to `add_anchored_child` and move to the section in the file called `STATIC FUNCTIONS` and add the function prototype to `STATIC PROTOTYPES`

liamHowatt commented on src/core/lv_obj_pos.c:
```diff
@@ -319,13 +319,26 @@ void lv_obj_align(lv_obj_t * obj, lv_align_t align, int32_t x_ofs, int32_t y_ofs
 
+static void add_anchored_children(lv_obj_t * obj, const lv_obj_t * base)
+{
+    base->spec_attr->anchored_child_count++;
```
Somewhere before we arrive here we need to do one of

- `if(base->spec_attr == NULL) return;`
- `lv_obj_allocate_spec_attr(base);`

May not be in this function, but somewhere, such that the test cases in the CI pass and there's no chance of it being NULL.

liamHowatt commented on examples/styles/lv_example_style_16.c:
```diff
@@ -0,0 +1,82 @@
+static lv_obj_t * obj_reference;
+
+/* Animate the reference object to bounce up & down. */
+static void lv_anim_callback(void * obj, int value)
```
`lv_` not needed

liamHowatt commented on examples/styles/lv_example_style_16.c:
```diff
@@ -0,0 +1,82 @@
+/* Animate the reference object to bounce up & down. */
+static void lv_anim_callback(void * obj, int value)
+{
+    (void) obj;
```
```suggestion
    LV_UNUSED(obj);
```

liamHowatt reviewed (commented):
Thanks very much @valeriosetti for sticking with this feature for so long and opening a PR!

After you respond to this feedback I will look deeper into the sustainability of your implementation.

<br>

Free the `anchored_children` [here](https://github.com/valeriosetti/lvgl/blob/997520a1648eeb522d9908a97a0927a6642cb063/src/core/lv_obj.c#L374). i.e.

```c
if(obj->spec_attr->anchored_children) {
    lv_free(obj->spec_attr->anchored_children);
    obj->spec_attr->anchored_children = NULL;
}
```

<br>

This test is failing because the behavior of `lv_obj_align_to` has changed. If you're able to make this function maintain compatibility, please do. Otherwise if this behavior is correct then update the ref image with the error one to become the new ref image.

tests/ref_imgs/libs/freetype_1.lp64.png
https://github.com/valeriosetti/lvgl/blob/anchor/tests/src/test_cases/libs/test_freetype.c#L450

Original:
![freetype_1 lp64](https://github.com/lvgl/lvgl/assets/30486941/f766024e-2fdf-4748-b25c-ab8d10b47b57)
Error:
![freetype_1 lp64_err](https://github.com/lvgl/lvgl/assets/30486941/18cabc32-2482-41b1-bffd-8f0571799855)

valeriosetti commented:
Thanks @liamHowatt for your review! I will start addressing comments.
@kisvegabor thanks for the feedback. I thought that #5665 (linked in the description) was enough for this. Should I open a new one?

kisvegabor commented:
My closing though in #5665 was
> Sorry, for mentioning only problems, however these are the main concerns why it is not implemented yet. I hope that we can figure out something together.

So I think it requires a deeper investigation to discover all options. Maybe we will find a completely different approach if we dive deeper. 


> Should I open a new one?

Yes, please.
